src/ui/ResultList/ResultList.ts

```diff
@@ -861,6 +861,18 @@ export class ResultList extends Component {
       this.element.id = uniqueId('coveo-result-list');
     }
   }
+
+  private enableBasedOnActiveLayouts() {
+    // We should try to enable a result list only when the result layout selector currently allow that result list layout to be displayed.
+    each(this.resultLayoutSelectors, resultLayoutSelector => {
+      const found = find(resultLayoutSelector.activeLayouts, (activeLayout, activeLayoutKey) => activeLayoutKey == this.options.layout);
+      if (found) {
+        super.enable();
+        resultLayoutSelector.enableLayouts([this.options.layout] as ValidLayout[]);
+        $$(this.element).removeClass('coveo-hidden');
```

Comment:
Should the loop stop at this point after enabling a found layout?

Reply:
It's there to handle the corner case of if you have multiple layout selector in the page.